### PR TITLE
TST/DOC: Fix tests and docs for .cat raising AttributeError if invalid

### LINKED ISSUE #8814: API: preferred way to check if column/Series has Categorical dtype
From http://stackoverflow.com/questions/26924904/check-if-dataframe-column-is-categorical/26925340#26925340

What is the preferred way to check for categorical dtype?

I now answered:

```
In [42]: isinstance(df.cat_column.dtype, pd.core.common.CategoricalDtype)
Out[42]: True

In [43]: pd.core.common.is_categorical_dtype(df.cat_column)
Out[43]: True
```

But:

- this seems somewhat buried in pandas. Should there be a more top-level function to do this?
- we should add the preferred way to the categorical docs.

### PR DESCRIPTION
**Pull request by shoyer**
Follow-up on #9617

Fixes #8814 -- is it fair to say that `hasattr(s, 'cat')` is probably
the best solution we're going to come up with for checking for categorical
data?

CC @jreback @jorisvandenbossche @JanSchulz

### CODE REVIEW

File: doc/source/categorical.rst [hunk 1]
```diff
@@ -766,6 +766,14 @@ Dtype comparisons work:
     dtype == np.str_
     np.str_ == dtype
 
+To check if a Series contains Categorical data, with pandas 0.16 or later, use
+``hasattr(s, 'cat')``:
+
+.. ipython:: python
```
Comment (jorisvandenbossche) [inline doc/source/categorical.rst]: can you add a `:okexcept:` option to the directive here (so the doc building does not issue a warning for the failure)?

Comment (shoyer) [inline doc/source/categorical.rst]: `hasattr` catches the error, so I don't think that's necessary?

Comment (jorisvandenbossche) [inline doc/source/categorical.rst]: ah, yes, sorry, I was looking too quickly, thought you were giving an example of that accessing the attribute raises, but it is `hasattr`, not `getattr` ..

**Comment by shoyer** [comment]
FYI this commit passed on my fork so we can safely merge this: https://travis-ci.org/shoyer/pandas/builds/53882855

**Comment by jreback** [comment]
ok, looks ok to me. go ahead and merge.

**Pull request merged**